server/sonar-db-dao/src/main/java/org/sonar/db/permission/AuthorizationDao.java

```diff
@@ -135,6 +135,17 @@ public int countUsersWithGlobalPermissionExcludingUserPermission(DbSession dbSes
       });
   }
 
+  public Set<String> keepAuthorizedProjectUuids(DbSession dbSession, Collection<String> projectUuids, @Nullable Integer userId, String permission) {
+    return executeLargeInputsIntoSet(
+      projectUuids,
+      partition -> {
+        if (userId == null) {
+          return mapper(dbSession).keepAuthorizedProjectUuidsForAnonymous(permission, projectUuids);
```

Comment:
Incredible, that means that there is the same bug in keepAuthorizedProjectIds() !